### DISCUSSION TITLE
Why would `CDeviceNameService::Get ()->GetDevice ("ukbd1", FALSE)` return 0?

### DISCUSSION THREAD

**Discussion by JoeStrout** [rsta2/circle#437] [Q&A] [answered]
I'm trying to access the keyboard on my RPi 400.  It works fine in the samples, but in my code, this call returns 0 (nullptr) every time.  Not just at startup; I try again every second or so, and it's always 0, even though my kernel is otherwise functional.

Any idea what could cause this?  Is there an initialization step I'm missing?

(My code is based on circle-stdlib; not sure if that matters.)

**Comment by rsta2** [comment, marked as answer]
From which base class have you derived your CKernel class? Use need at least 
CStdlibAppStdio (or CStdlibAppNetwork), when working with circle-stdlib, to 
have USB support in the system. The keyboard cannot work without. Also 
Initialize() of this base class must be called, in case you did override 
Initialize().

**Reply by JoeStrout** [reply]
That's it!  I'm derived from CStdlibAppStudio, but I had failed to call the base class Initialize().  Thank you!
